cosmwasm/main.go

```diff
@@ -0,0 +1,280 @@
+package main
+
+import (
+	"embed"
+	"encoding/gob"
+	"fmt"
+	"io"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	hplugin "github.com/hashicorp/go-plugin"
+
+	"github.com/ignite/cli/ignite/services/plugin"
+)
+
+//go:embed wasm-wiring/*
+var templates embed.FS // Embedded template files
+
+func init() {
+	gob.Register(plugin.Manifest{})
+	gob.Register(plugin.ExecutedCommand{})
+	gob.Register(plugin.ExecutedHook{})
+}
+
+type p struct {
+	chainName string
+}
+
+// NewPlugin creates a new plugin instance with the given chainName and chainNamed values.
+func NewPlugin(chainName string) *p {
+	return &p{
+		chainName: chainName,
+	}
+}
+
+// ChainName returns the chainName of the plugin.
+func (p *p) ChainName() string {
+	return p.chainName
+}
+
+// ChainNamed derives and returns the chainNamed of the plugin.
+func (p *p) ChainNamed() string {
+	return p.chainName + "d"
+}
+func (p) Manifest() (plugin.Manifest, error) {
+	return plugin.Manifest{
+		Name: "cosmwasm",
+		// Add commands here
+		Commands: []plugin.Command{
+			// Example of a command
+			{
+				Use:               "cosmwasm",
+				Short:             "The cosmwasm command is used for adding cosmwasm support for apps scaffolded with Ignite CLI",
+				PlaceCommandUnder: "ignite",
+				// Examples of adding subcommands:
+				Commands: []plugin.Command{
+					{
+						Use:               "add",
+						Short:             "Add a CosmWasm module to the chain",
+						Long:              "This command will install dependencies, create new files, and modify existing files to add a CosmWasm module to the chain.",
+						PlaceCommandUnder: "ignite cosmwasm",
+					},
+				},
+			},
+		},
+	}, nil
+}
+
+func (p *p) Execute(cmd plugin.ExecutedCommand) error {
+
+	// According to the number of declared commands, you may need a switch:
+
+	switch cmd.Use {
+	case "add":
+		return handleAddCommand(p)
+	default:
+		return fmt.Errorf("unknown command: %s", cmd.Use)
+	}
+}
+
+func handleAddCommand(p *p) error {
+	fmt.Println("Adding stuff...")
+
+	err := installDependencies()
+	if err != nil {
+		return err
+	}
+
+	err = createNewFiles(p)
+	if err != nil {
+		return err
+	}
+
+	//err = modifyExistingFiles()
+	//if err != nil {return err}
+
+	return nil
+}
```

Comment:
A couple of things within this suggestion:

 - The `handleAddCommand()` function would make sense as a method of plugin
 - Something like "Updating app files..." would be more informative
 - Sending `p.chainName` to `createNewFiles()` would simplify the function

```suggestion
func (p p) handleAddCommand() error {
	fmt.Println("Updating app files...")

	err := installDependencies()
	if err != nil {
		return err
	}

	err = createNewFiles(p.chainName)
	if err != nil {
		return err
	}

	return nil
}
```

Reply:
Resolved with https://github.com/yerasyla/plugins/commit/27f2900b6c853a13a1269fa3d74fe471399bc399 and https://github.com/yerasyla/plugins/commit/fc9fc29ba5417fda941bd2209f3fb7b6cadfe9b8